models/articles.models.js

```diff
@@ -0,0 +1,26 @@
+const db = require("../db/connection");
+
+exports.insertArticleById = (id) => {
+  return db
+    .query(`SELECT * FROM articles WHERE articles.article_id = $1;`, [id])
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, message: "Not Found" });
+      }
+      return rows[0];
+    });
+};
+
+exports.insertArticles = () => {
+  return db
+    .query(
+      `
+  SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY articles.created_at DESC;`
+    )
+    .then(({ rows }) => {
+      rows.forEach((article) => {
+        article.comment_count = JSON.parse(article.comment_count);
```

Comment:
I'd perhaps think of a more explicit way to change comment_count into a number, I think you can use something like CAST(comment_count AS INT) in SQL :)